util/stringutil/string_util_test.go

```diff
@@ -152,3 +152,19 @@ func (s *testStringUtilSuite) TestIsExactMatch(c *C) {
 		c.Assert(IsExactMatch(patTypes), Equals, v.exactMatch, Commentf("%v", v))
 	}
 }
+
+func BenchmarkMatchSpecial(b *testing.B) {
+	var (
+		pattern = `a%a%a%a%a%a%a%a%b`
+		target  = `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa`
+		escape  = byte('\\')
+	)
+
+	patChars, patTypes := CompilePattern(pattern, escape)
+
+	b.StartTimer()
+	match := DoMatch(target, patChars, patTypes)
+	if match {
+		b.Fatal("Unmatch expected.")
+	}
```

Comment:
The benchmark usage is wrong. Have a try using following code. And address the benchmark result in the description.
```
for i := 0;i < b.N;i++ {
    match := DoMatch(target, patChars, patTypes)
    if match {
        b.Fatal("Unmatch expected.")
    }
}
```